### ISSUE TITLE
TextArea: property "text" not updated

### ISSUE THREAD

**Issue by moodyhunter** [OlivierLDff/Qaterial#114]
See example https://tinyurl.com/ydorg5dq where the button text should follow the TextArea content.
![image](https://user-images.githubusercontent.com/76251897/124603936-04ff3d00-de9d-11eb-8d7c-e812d201b8df.png)

- Commenting out the `import Qaterial` line and run without Qaterial does not have this problem: 

![image](https://user-images.githubusercontent.com/76251897/124605393-6f64ad00-de9e-11eb-9008-23bd54e309e3.png)

**Comment by OlivierLDff** [comment]
Big issue of qaterial text area is that it doesn't herit from Template.Textarea but act like a proxy
This should be fixed in future version when I will have time (I'm scarred to break existing code)

**Comment by OlivierLDff** [comment]
Best would be to have a TextArea not really styled (only font, no background) and have a TextArea2014, TextAreaOutlined, TextAreaFilled that bring a title and a background

**Comment by moodyhunter** [comment]
Thanks for the explanation.
BTW, is there any replacement for the multiline text input?

**Comment by OlivierLDff** [comment]
Quick fix would be here : https://github.com/OlivierLDff/Qaterial/blob/ff61e9808a96b85e018d17d48e0056a7dce76578/qml/Qaterial/TextArea.qml#L375-L405

To add `onTextChanged: () => _control.text = text`

**Comment by moodyhunter** [comment]
Thanks, I'll try it out and hope to see your refactor going online!

**Comment by OlivierLDff** [comment]
If you want my general opinion about how I did Qaterial right now, after some usage time:
- I feel that Qaterial basic Controls are doing more than they should
- It is hard to customize basic components (because Qaterial already customize too much basic components)

You can see that my latest commit were about very simple items:
IconLabelWithCaption
![](https://user-images.githubusercontent.com/17255804/118284096-5bb94d80-b4d0-11eb-9b7a-306f1036397d.png)
LabelWithCaption
![](https://user-images.githubusercontent.com/17255804/118148212-a8d5ea80-b410-11eb-810d-279da905bbe9.png)
Icon
![](https://user-images.githubusercontent.com/17255804/118131081-5be91880-b3fe-11eb-91cf-ba8179c5446b.png)

That provide really useful and reusable building block for complex component.

I realized it is easier to create custom items that meet specific need based on very simple Controls, than trying to customize a Control that can fit them all and become a Frankenstein Item bloated everywhere and over complex (ie current TextField, RawMaterialButton, etc...) 

So my future update will go in that direction. Keep every `Control` really basic, and provide some decorated variant following Material Design documentation.
And basic `Control` shoudn't add custom properties, otherwise consistency is too hard to follow.

To customize colors, I will use `palette` that is there everywhere (even if it is not mean for that).
This will be easier for me to keep consistency (and make color customization easier).

**Comment by OlivierLDff** [comment]
I merged your quick fix in https://github.com/OlivierLDff/Qaterial/pull/123
Real fix would be https://github.com/OlivierLDff/Qaterial/issues/88
When I will have time for a future breaking version of Qaterial

**Comment by moodyhunter** [comment]
thanks!
